### PR TITLE
test(cli): add monorepo e2e test based on a custom fixture

### PR DESCRIPTION
**Pull request by byCedric**
# Why

This is a draft to add a specific monorepo test to the exporting e2e tests.

> Note, I pulled out the package linking logic. As discussed, we are probably going for the tarball option as it's closer to how packages are being published and keeps things better isolated. But I'll do that in a follow-up PR.

# How

- We need a custom fixture for this, so I added one based on `e2e/fixtures/with-router`
- Add a single test case that checks if both projects are not leaking code, based on static routes

# Test Plan

- Test without the fix is confirmed failing in: https://github.com/expo/expo/actions/runs/9499559787/job/26180698726?pr=29735#step:10:115
- When rebasing on the fix from #29733, it should work

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by byCedric** [comment]
Ah yeah, the fixture is not connected to the source code, hence it's still failing.

File: packages/@expo/cli/e2e/fixtures/with-router-monorepo/apps/app-b/app.config.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/** @type {import('expo/config').ExpoConfig} */
+module.exports = {
+  scheme: 'acme',
+  android: {
+    package: 'com.example.minimal.appb',
+  },
+  ios: {
+    bundleIdentifier: 'com.example.minimal.appb',
+  },
```
Comment (EvanBacon) [inline packages/@expo/cli/e2e/fixtures/with-router-monorepo/apps/app-b/app.config.js:9]: trim back fixtures

File: packages/@expo/cli/e2e/fixtures/with-router-monorepo/apps/app-b/babel.config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = function (api) {
```
Comment (EvanBacon) [inline packages/@expo/cli/e2e/fixtures/with-router-monorepo/apps/app-b/babel.config.js:1]: don't need this

File: packages/@expo/cli/e2e/fixtures/with-router-monorepo/apps/app-a/app/_layout.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Stack } from 'expo-router';
```
Comment (EvanBacon) [inline packages/@expo/cli/e2e/fixtures/with-router-monorepo/apps/app-a/app/_layout.js:1]: drop this file

**Review by EvanBacon — APPROVED** [review]

**Comment by byCedric** [comment]
The `babel-preset-expo` test failure is unrelated to this test.

**Pull request merged**